src/purchase-selection/utils.ts

```diff
@@ -0,0 +1,155 @@
+import {PreassignedFareProduct} from '@atb/configuration/types';
+import {compareVersion} from '@atb/utils/compare-version';
+import type {
+  PurchaseSelectionBuilderInput,
+  PurchaseSelectionType,
+} from './types';
+import {TariffZoneWithMetadata} from '@atb/tariff-zones-selector';
+import turfBooleanPointInPolygon from '@turf/boolean-point-in-polygon';
+import {UserProfileWithCount} from '@atb/fare-contracts';
+import {TariffZone, UserProfile} from '@atb-as/config-specs';
+import {isValidDateString} from '@atb/utils/date';
+
+export const isProductSellableInApp = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+) => {
+  if (
+    (product.limitations.appVersionMin &&
+      compareVersion(product.limitations.appVersionMin, input.appVersion) >
+        0) ||
+    (product.limitations.appVersionMax &&
+      compareVersion(product.limitations.appVersionMax, input.appVersion) < 0)
+  )
+    return false;
+
+  if (
+    product.distributionChannel.some((channel) => channel === 'debug-app') &&
+    input.customerProfile?.debug
```

Comment:
I understand that this is used for testing but should not this belong to the test itself not as part of the utils?

Reply:
I think this is related to tickets that are sold to the distribution channel of `debug-app`, and also only allowed for users marked with `debug`. This is mainly here to allow testing new tickets on prod, such as when Gunn needs to test something.

Reply:
Yeah, that's correct. Even though the code mentions 'debug' and its purpose is to allow testing, it it still production code.

Reply:
It is good to refresh this concept, I was a bit confused.